dali/operators/math/expressions/broadcasting_test.cc

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include "dali/operators/math/expressions/broadcasting.h"
+#include "dali/core/tensor_shape.h"
+#include "dali/kernels/common/utils.h"
+
+namespace dali {
+namespace test {
+
+TEST(ArithmeticOpsBroadcastingTest, BroadcastShape) {
+  auto test = [](TensorShape<> expected, TensorShape<> a, TensorShape<> b) {
+    TensorShape<> result0, result1;
+    ASSERT_TRUE(CanBroadcast(a, b));
+    ASSERT_TRUE(CanBroadcast(b, b));
```

Comment:
Below you check {b, a} for the TensorListShape in the second check. Just a note.